client/me/reauth-required/index.jsx

```diff
@@ -268,7 +268,11 @@ const ReauthRequired = createReactClass( {
 				<TwoFactorActions
 					twoFactorAuthType={ twoFactorAuthType }
 					onChange={ this.handleAuthSwitch }
-					isSmsSupported={ method === 'sms' || method === 'authenticator' }
+					isSmsSupported={
+						method === 'sms' ||
+						( method === 'authenticator' &&
+							this.props.twoStepAuthorization.data.two_step_sms_last_four.length )
```

Comment:
@xknown I added this condition to show the "Send code via text message" button only when an SMS number has been added. When there is no SMS second factor (like for an a8c account), there's no second method shown:
<img width="426" alt="Screen Shot 2020-08-12 at 1 35 42 PM" src="https://user-images.githubusercontent.com/1689238/90069229-042b4e80-dcc0-11ea-96a1-bcc9d8dae2d1.png">

Does that fix the issue while keeping the SMS fallback option around for people who enabled it?